arch/arm/include/stm32/stm32g47xxx_irq.h

```diff
@@ -0,0 +1,204 @@
+
+/* This file should never be included directly but, rather, only indirectly
+ * through nuttx/irq.h
+ */
+
+#ifndef __ARCH_ARM_INCLUDE_STM32_STM32G47XXX_IRQ_H
+#define __ARCH_ARM_INCLUDE_STM32_STM32G47XXX_IRQ_H
+
+/****************************************************************************************************
+ * Included Files
+ ****************************************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/irq.h>
+
+/****************************************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************************************/
+
+/* IRQ numbers.  The IRQ numbers correspond to the vector numbers and hence
+ * map directly to bits in the NVIC.  This does, however, waste several words
+ * of memory in the IRQ to handle mapping tables.
+ *
+ * Processor Exceptions (vectors 0-15) are common to all STM32 parts are are
+ * found in nuttx/arch/arm/include/stm32/irq.h.  They are not repeated here.
+ *
+ * Other interrupts (vectors >= 16) are defined below.
+ */
+
+#define STM32_IRQ_WWDG        (STM32_IRQ_FIRST+0)   /* 0:  Window Watchdog interrupt */
+#define STM32_IRQ_PVD         (STM32_IRQ_FIRST+1)   /* 1:  PVD through EXTI Line detection interrupt */
+#define STM32_IRQ_TAMPER      (STM32_IRQ_FIRST+2)   /* 2:  Tamper interrupt, or Time Stamp (see STM32_IRQ_TIMESTAMP) */
+#define STM32_IRQ_RTC_WKUP    (STM32_IRQ_FIRST+3)   /* 3:  RTC global interrupt */
+#define STM32_IRQ_FLASH       (STM32_IRQ_FIRST+4)   /* 4:  Flash global interrupt */
+#define STM32_IRQ_RCC         (STM32_IRQ_FIRST+5)   /* 5:  RCC global interrupt */
+#define STM32_IRQ_EXTI0       (STM32_IRQ_FIRST+6)   /* 6:  EXTI Line 0 interrupt */
+#define STM32_IRQ_EXTI1       (STM32_IRQ_FIRST+7)   /* 7:  EXTI Line 1 interrupt */
+#define STM32_IRQ_EXTI2       (STM32_IRQ_FIRST+8)   /* 8:  EXTI Line 2 interrupt, or */
+#define STM32_IRQ_EXTI3       (STM32_IRQ_FIRST+9)   /* 9:  EXTI Line 3 interrupt */
+
+#define STM32_IRQ_EXTI4       (STM32_IRQ_FIRST+10)  /* 10: EXTI Line 4 interrupt */
+#define STM32_IRQ_DMA1CH1     (STM32_IRQ_FIRST+11)  /* 11: DMA1 channel 1 global interrupt */
+#define STM32_IRQ_DMA1CH2     (STM32_IRQ_FIRST+12)  /* 12: DMA1 channel 2 global interrupt */
+#define STM32_IRQ_DMA1CH3     (STM32_IRQ_FIRST+13)  /* 13: DMA1 channel 3 global interrupt */
+#define STM32_IRQ_DMA1CH4     (STM32_IRQ_FIRST+14)  /* 14: DMA1 channel 4 global interrupt */
+#define STM32_IRQ_DMA1CH5     (STM32_IRQ_FIRST+15)  /* 15: DMA1 channel 5 global interrupt */
+#define STM32_IRQ_DMA1CH6     (STM32_IRQ_FIRST+16)  /* 16: DMA1 channel 6 global interrupt */
+#define STM32_IRQ_DMA1CH7     (STM32_IRQ_FIRST+17)  /* 17: DMA1 channel 7 global interrupt */
+#define STM32_IRQ_ADC1        (STM32_IRQ_FIRST+18)  /* 18: ADC1 and ADC2 global interrupt (see also STM32_IRQ_ADC2) */
```

Comment:
This should be called STM32_IRQ_ADC12. Otherwise, it will not work with the current ADC implementation.

Reply:
Good catch! Fixed in db33458048443413ea95313d3959ace7fb6e5183. Thank you.